jsonrpc/server.go

```diff
@@ -435,10 +444,13 @@ func (s *Server) handleRequest(ctx context.Context, req *Request) (*response, er
 			reqJSON, _ := json.Marshal(req)
 			errJSON, _ := json.Marshal(res.Error)
 			s.log.Debugw("Failed handing RPC request", "req", string(reqJSON), "res", string(errJSON))
+			s.log.Tracew("Internal error during RPC handling", "req", string(reqJSON), "error", string(errJSON))
 		}
 		return res, nil
 	}
 	res.Result = tuple[0].Interface()
+	resJSON, _ := json.Marshal(res) // Serialise response for logging
```

Comment:
This will marshal for all log levels, but only log when the level=trace. I wonder if we can modify this so that we only cal json.Marshal when the log level is trace. Or just drop this log?

Reply:
Wdyt @kirugan @IronGauntlets

Reply:
Yes we need to marshal at the top and here only in case trace level is enabled. @KananGiovanni please note that we need to add request/response information to the log only -
there is no need to add additional `s.log.Tracew(...)` in this method.

Reply:
> I wonder if we can modify this so that we only call json.Marshal when the log level is trace

We can add method to check current log level in Logger (it's already there in ZapLogger, we will need to modify interface and probably wrapper)